snapcraft/internal/os_release.py

```diff
@@ -38,12 +38,15 @@ def __init__(self, *, os_release_file: str = '/etc/os-release') -> None:
 
         :param str os_release_file: Path to os-release file to be parsed.
         """
-        with open(os_release_file) as f:
-            self._os_release = {}  # type: Dict[str, str]
-            for line in f:
-                entry = line.rstrip().split('=')
-                if len(entry) == 2:
-                    self._os_release[entry[0]] = entry[1].strip('"')
+        self._os_release = {}  # type: Dict[str, str]
+        try:
+            with open(os_release_file) as f:
+                for line in f:
+                    entry = line.rstrip().split('=')
+                    if len(entry) == 2:
+                        self._os_release[entry[0]] = entry[1].strip('"')
+        except FileNotFoundError:
+            pass
```

Comment:
I kinda prefer `contextlib.suppress` for the type of operation that we know may toss an exception that we know we want to ignore.